libs/sdk-ui-ext/src/internal/components/configurationControls/colors/colorDropdown/tests/ColorPalette.test.tsx

```diff
@@ -15,55 +15,52 @@ const defaultProps: IColorPaletteProps = {
+        ...render(<ColorPalette {...props} />),
+    };
 }
 
 describe("ColorPalette", () => {
-    it("should render ColorPalette control", () => {
-        const wrapper = createComponent();
-        expect(wrapper.find(ColorPalette).length).toBe(1);
-    });
+    afterEach(cleanup);
 
     it("should render small ColorPalette control", () => {
-        const wrapper = createComponent();
-        expect(wrapper.find("div").first().hasClass("gd-color-drop-down-list")).toBeTruthy();
-        expect(wrapper.find("div").first().hasClass("gd-color-drop-down-list-large")).toBeFalsy();
+        const { getAllByLabelText } = createComponent();
+        expect(getAllByLabelText(/rgb*/i).length).toBe(5);
     });
 
     it("should render large ColorPalette control", () => {
-        const wrapper = createComponent({ colorPalette: getLargePalette() });
-        expect(wrapper.find("div").first().hasClass("gd-color-drop-down-list")).toBeFalsy();
-        expect(wrapper.find("div").first().hasClass("gd-color-drop-down-list-large")).toBeTruthy();
+        const { getAllByLabelText } = createComponent({ colorPalette: getLargePalette() });
+        expect(getAllByLabelText(/rgb*/i).length).toBe(20);
     });
 
-    it("should render 5 ColorPaletteItem controls", () => {
-        const wrapper = createComponent();
-        expect(wrapper.find(ColorPaletteItem).length).toBe(5);
+    it("should render 5 ColorPaletteItem controls one has to be selected", () => {
+        const { fill, guid } = colorPalette[3];
+        const { getByLabelText } = createComponent({ selectedColorGuid: guid });
+        expect(getByLabelText(`rgb(${fill.r},${fill.g},${fill.b})`)).toHaveClass("gd-color-list-item-active");
     });
 
     it("should render 5 ColorPaletteItem controls any has to be selected", () => {
-        const wrapper = createComponent();
-        const selectedItem = wrapper.find(ColorPaletteItem).find({ selected: true });
-        expect(selectedItem.length).toBe(0);
-    });
-
-    it("should render 5 ColorPaletteItem controls one has to be selected", () => {
-        const wrapper = createComponent({ selectedColorGuid: "04" });
-        const selectedItem = wrapper.find(ColorPaletteItem).find({ selected: true });
-        expect(selectedItem.length).toBe(1);
+        const { container } = createComponent();
+        expect(container.getElementsByClassName("gd-color-list-item-active").length).toBe(0);
     });
 
     it("should render 5 ColorPaletteItem controls any to be selected by fake guid", () => {
-        const wrapper = createComponent({ selectedColorGuid: "fakegid" });
-        const selectedItem = wrapper.find(ColorPaletteItem).find({ selected: true });
-        expect(selectedItem.length).toBe(0);
+        const { container } = createComponent({ selectedColorGuid: "fakegid" });
+        expect(
+            container.getElementsByClassName("gd-color-list-item-active s-color-list-item-fakegid").length,
+        ).toBe(0);
```

Comment:
Does not this test the same thing as the previous test?

Reply:
I have modified the test title to be clearer.